src/nspawn/nspawn.c

```diff
@@ -2988,6 +2988,23 @@ static int on_request_stop(sd_bus_message *m, void *userdata, sd_bus_error *erro
 static int determine_names(void) {
         int r;
 
+        /* Find newest version in case directory ends with "auto.d", but not if we found it with machine */
+        if (arg_image) {
+                char *arg_image_unresolved = arg_image;
```

Comment:
we use the `arg_` prefix so far only for global variables that are directly process parameters (i.e. parsed from the cmdline). Please name these local variables differently to avoid any confusion.